webapp/graphite/render/views.py

```diff
@@ -173,6 +173,34 @@ def renderView(request):
         add_never_cache_headers(response)
       return response
 
+    if format == 'json_dygraph':
+      labels = ['Time']
+      datapoints = [[ts] for ts in range(data[0].start, data[0].end, data[0].step)]
+      for series in data:
+        labels.append(series.name)
+        for i, point in enumerate(series):
+          datapoints[i].append(point if point is not None else 'null')
+      line_template = '[%%s000%s]' % ''.join([', %s'] * len(data))
+      lines = [line_template % tuple(points) for points in datapoints]
+      result = '{"labels" : %s, "data" : [%s]}' % (json.dumps(labels), ', '.join(lines))
```

Comment:
That's a rather odd way to generate JSON. Could you refactor that to read `result = json.dums(some_structure)` with native types (lists / dicts) rather than string interpolation?